frame/contracts/src/storage.rs

```diff
@@ -183,13 +183,14 @@ where
 				child::kill_storage(&child_trie_info(&trie.trie_id), Some(remaining_key_budget));
 			let keys_removed = match outcome {
 				// This should not happen as our budget was large enough to remove all keys.
-				KillStorageResult::SomeRemaining(count) => count,
-				KillStorageResult::AllRemoved(count) => {
+				KillStorageResult::SomeRemaining { backend: count, overlay: 0 } => count,
+				KillStorageResult::AllRemoved { backend: count, overlay: 0 } => {
 					// We do not care to preserve order. The contract is deleted already and
 					// noone waits for the trie to be deleted.
 					queue.swap_remove(0);
 					count
 				},
+				_ => 0
```

Comment:
overlay should always be 0 in this case, but in case it is not I think it is better to still return the number of key removed from backend.
```suggestion
				// Note: no keys should be removed from the overlay
				KillStorageResult::SomeRemaining { backend: count, overlay: _ } => count,
				// Note: no keys should be removed from the overlay
				KillStorageResult::AllRemoved { backend: count, overlay: _ } => {
					// We do not care to preserve order. The contract is deleted already and
					// noone waits for the trie to be deleted.
					queue.swap_remove(0);
					count
				},
```

Reply:
Oh...ok